packages/components/src/ui/data-table-filter/lib/memo.ts

```
@@ -0,0 +1,35 @@
+    const deps = getDeps()
+
+    // If no previous deps or deps have changed, recompute
+    if (!prevDeps || !shallowEqual(prevDeps, deps)) {
+      // console.log(`[memo] Cache MISS - ${options.key}`)
+      cachedResult = compute(deps)
+      prevDeps = deps
+    } else {
+      // console.log(`[memo] Cache HIT - ${options.key}`)
+    }
+
+    return cachedResult!
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)